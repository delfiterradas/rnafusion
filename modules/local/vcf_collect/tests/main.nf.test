nextflow_process {

    name "Test Process VCF_COLLECT"
    script "../main.nf"
    process "VCF_COLLECT"
    config './nextflow.config'

    // setup {
    //     run("STARFUSION_BUILD") {
    //         script "../../../nf-core/starfusion/build/main.nf"
    //         process {
    //             """
    //             input[0] = [
    //                 [ id:'minigenome_fasta' ],
    //                 file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/minigenome.fa')
    //             ]
    //             input[1] = [
    //                 [ id:'minigenome_gtf' ],
    //                 file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/minigenome.gtf')
    //             ]
    //             input[2] = file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/CTAT_HumanFusionLib.mini.dat.gz')
    //             input[3] = "homo_sapiens"
    //             input[4] = "http://ftp.ebi.ac.uk/pub/databases/Pfam/releases/Pfam37.4/Pfam-A.hmm.gz"
    //             input[5] = Channel.of([
    //                 "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm",
    //                 "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3f",
    //                 "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3i",
    //                 "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3m",
    //                 "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3p"
    //             ])
    //             input[6] = "https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/AnnotFilterRule.pm"
    //             """
    //         }
    //     }

    //     run("FUSIONINSPECTOR") {
    //         script "../../fusioninspector/main.nf"
    //         process {
    //             """
    //             input[0] = Channel.of([
    //                 [ id:'test', single_end:false ], // meta map
    //                 [
    //                     file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/v0.0.1/rnaseq_1.fastq.gz"),
    //                     file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/v0.0.1/rnaseq_2.fastq.gz")
    //                 ], // reads
    //                 [file("https://github.com/FusionInspector/FusionInspector/raw/master/test/fusion_targets.A.txt")]
    //             ])
    //             input[1] = STARFUSION_BUILD.out.reference
    //             """
    //         }
    //     }

    //     run("FUSIONREPORT_DOWNLOAD") {
    //         script "../../../nf-core/fusionreport/download/main.nf"
    //         process {
    //             """
    //             """
    //         }
    //     }

    //     run("FUSIONREPORT_DETECT") {
    //         script "../../../nf-core/fusionreport/detect/main.nf"
    //         process {
    //             """
    //             input[0] = Channel.of([
    //                 [ id:'test_sample' ], // meta map
    //                 file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/arriba.tsv', checkIfExists: true),
    //                 file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/starfusion.tsv', checkIfExists: true),
    //                 file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/fusioncatcher.txt', checkIfExists: true)
    //             ])
    //             input[1] = FUSIONREPORT_DOWNLOAD.out.fusionreport_ref
    //             input[2] = 1
    //             """
    //         }
    //     }
    // }

    test("VCF_COLLECT - human - fusionreport - fusioninspector") {
        config './nextflow.config'

        setup {
            run("STARFUSION_BUILD") {
                script "../../../nf-core/starfusion/build/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'minigenome_fasta' ],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/minigenome.fa')
                    ]
                    input[1] = [
                        [ id:'minigenome_gtf' ],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/minigenome.gtf')
                    ]
                    input[2] = file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/CTAT_HumanFusionLib.mini.dat.gz')
                    input[3] = "homo_sapiens"
                    input[4] = "http://ftp.ebi.ac.uk/pub/databases/Pfam/releases/Pfam37.4/Pfam-A.hmm.gz"
                    input[5] = Channel.of([
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3f",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3i",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3m",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3p"
                    ])
                    input[6] = "https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/AnnotFilterRule.pm"
                    """
                }
            }

            run("FUSIONINSPECTOR") {
                script "../../fusioninspector/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id:'test', single_end:false ], // meta map
                        [
                            file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/v0.0.1/rnaseq_1.fastq.gz"),
                            file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/v0.0.1/rnaseq_2.fastq.gz")
                        ], // reads
                        [file("https://github.com/FusionInspector/FusionInspector/raw/master/test/fusion_targets.A.txt")]
                    ])
                    input[1] = STARFUSION_BUILD.out.reference
                    """
                }
            }
        }

        when {
            process {
                """
                // def tsv_output = FUSIONINSPECTOR.out.tsv ?: Channel.empty()
                // def gtf_output = FUSIONINSPECTOR.out.out_gtf ?: Channel.empty()

                // // Use ifEmpty to provide fallback files
                // def tsv_file = tsv_output
                //     .map { meta, tsv -> tsv }
                //     .ifEmpty(file("dummy.tsv"))
                //     .first()

                // def gtf_file = gtf_output
                //     .map { meta, gtf -> gtf }
                //     .ifEmpty(file("dummy2.gtf"))
                //     .first()
                // fusioninspector_out = Channel.of([ id:'test_sample' ])
                //     .combine(FUSIONINSPECTOR.out.tsv.map { it[1] })
                //     .combine(FUSIONINSPECTOR.out.out_gtf.map { it[1] })
                // input[0] = [
                //     fusioninspector_out,
                //     file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/fusionreport/test_sample_fusionreport_index.html", checkIfExists: true),
                //     file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/fusionreport/test_sample.fusions.csv", checkIfExists: true)
                // ]
                // Create dummy files locally
                touch dummy.tsv
                touch dummy.gtf
                touch dummy.html
                touch dummy.csv
                touch dummy_hgnc.txt
                touch dummy_timestamp.txt

                input[0] = [
                    [ id:'test_sample' ],
                    file('dummy.tsv'),
                    file('dummy.gtf'),
                    file('dummy.html'),
                    file('dummy.csv')
                ]
                // input[0] = [
                //     [ id:'test_sample' ],
                //     tsv_file,
                //     gtf_file,
                //     file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/fusionreport/test_sample_fusionreport_index.html"),
                //     file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/fusionreport/test_sample.fusions.csv")
                // ]
                input[1] = [[ id:'hgnc' ], file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/reference/hgnc/hgnc_complete_set.txt", checkIfExists: true)]
                input[2] = [[ id:'hgnc_timestamp' ], file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/reference/hgnc/HGNC-DB-timestamp.txt", checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("VCF_COLLECT - human - fusionreport - fusioninspector - stub") {

        options "-stub"
        config './nextflow.config'

        setup {
            run("STARFUSION_BUILD") {
                script "../../../nf-core/starfusion/build/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'minigenome_fasta' ],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/minigenome.fa')
                    ]
                    input[1] = [
                        [ id:'minigenome_gtf' ],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/minigenome.gtf')
                    ]
                    input[2] = file('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/genome/CTAT_HumanFusionLib.mini.dat.gz')
                    input[3] = "homo_sapiens"
                    input[4] = "http://ftp.ebi.ac.uk/pub/databases/Pfam/releases/Pfam37.4/Pfam-A.hmm.gz"
                    input[5] = Channel.of([
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3f",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3i",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3m",
                        "https://www.dfam.org/releases/Dfam_3.8/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3p"
                    ])
                    input[6] = "https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/AnnotFilterRule.pm"
                    """
                }
            }

            run("FUSIONINSPECTOR") {
                script "../../fusioninspector/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id:'test', single_end:false ], // meta map
                        [
                            file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/v0.0.1/rnaseq_1.fastq.gz"),
                            file("https://github.com/STAR-Fusion/STAR-Fusion-Tutorial/raw/v0.0.1/rnaseq_2.fastq.gz")
                        ], // reads
                        [file("https://github.com/FusionInspector/FusionInspector/raw/master/test/fusion_targets.A.txt")]
                    ])
                    input[1] = STARFUSION_BUILD.out.reference
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test_sample' ],
                    FUSIONINSPECTOR.out.tsv[0][1],
                    FUSIONINSPECTOR.out.out_gtf[0][1],
                    file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/fusionreport/test_sample_fusionreport_index.html"),
                    file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/fusionreport/test_sample.fusions.csv")
                ]
                input[1] = [[ id:'hgnc' ], file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/reference/hgnc/hgnc_complete_set.txt")]
                input[2] = [[ id:'hgnc_timestamp' ], file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/reference/hgnc/HGNC-DB-timestamp.txt")]
                """
            }
            params {
                arriba                     = true
                starfusion                 = true
                fusioncatcher              = true
                no_cosmic                  = true
                outdir                     = "$outputDir"
                fusionreport_download_args = '--no-cosmic'
                fusionreport_detect_args   = {params.no_cosmic} ? "--no-cosmic" : ""
                fusionreport_detect_args2  = "--export csv"
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
