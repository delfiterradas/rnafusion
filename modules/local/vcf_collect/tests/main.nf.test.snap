{
    "VCF_COLLECT - human - fusionreport - fusioninspector - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "vcf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-22T18:38:55.111141883"
    },
    "VCF_COLLECT - human - fusionreport - fusioninspector": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "vcf": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-22T18:39:55.640372223"
    }
}